cpp/src/utility/bbox_thrust.cuh

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <thrust/functional.h>
+#include <thrust/pair.h>
+
+#include <ostream>
+
+#include "utility/z_order.cuh"
+
+namespace {
+
+template <typename T>
+using SBBox = thrust::pair<thrust::tuple<T, T>, thrust::tuple<T, T>>;
```

Comment:
Is it particularly useful to make this a pair of pairs?
```suggestion
using SBBox = thrust::tuple<T, T, T, T>;
```
You could add helper methods like
```
template<typename T> constexpr T x_min(SBBox bbox) { return thrust::get<0>(bbox); }
template<typename T> constexpr T x_max(SBBox bbox) { return thrust::get<1>(bbox); }
template<typename T> constexpr T y_min(SBBox bbox) { return thrust::get<2>(bbox); }
template<typename T> constexpr T y_max(SBBox bbox) { return thrust::get<3>(bbox); }
```

Reply:
This type isn't used anymore, but I've kept it here in case the other related in-flight PRs use it.